core/src/main/java/jenkins/security/ResourceDomainConfiguration.java

```diff
@@ -112,14 +114,14 @@ private FormValidation checkUrl(String resourceRootUrlString, boolean allowOnlin
 
         URL resourceRootUrl;
         try {
-            resourceRootUrl = new URL(resourceRootUrlString);
-        } catch (MalformedURLException ex) {
+            resourceRootUrl = new URI(resourceRootUrlString).toURL();
+        } catch (MalformedURLException | URISyntaxException ex) {
             return FormValidation.error(Messages.ResourceDomainConfiguration_Invalid());
         }
 
         String resourceRootUrlHost = resourceRootUrl.getHost();
         try {
-            String jenkinsRootUrlHost = new URL(jenkinsRootUrlString).getHost();
+            String jenkinsRootUrlHost = new URI(jenkinsRootUrlString).toURL().getHost();
```

Comment:
the host is exposed from the URI?
```suggestion
            String jenkinsRootUrlHost = new URI(jenkinsRootUrlString).getHost();
```